eng/scripts/verify-npm-tags.ps1

```diff
@@ -0,0 +1,50 @@
+param (
+  [Parameter(mandatory = $true)]
+  $originalDistTags,
+  [Parameter(mandatory = $true)]
+  $intendedTag,
+  [Parameter(mandatory = $true)]
+  $intendedTagVersion,
+  [Parameter(mandatory = $true)]
+  $packageJson,
+  [Parameter(mandatory = $true)]
+  $npmToken
+)
+
+$ErrorActionPreference = 'Stop'
+$PSNativeCommandUseErrorActionPreference = $true
+
+$pkg = Get-Content -Raw $packageJson | ConvertFrom-Json
+$packageName = $pkg.Name
+
+Write-Host "Verify npm tag versions for package $packageName"
+
+$parsedOriginalDistTags = $originalDistTags | ConvertFrom-Json
+
+$npmPkgProp = npm view $packageName --json | ConvertFrom-Json
+$packageDistTags = $npmPkgProp."dist-tags"
+
+Write-Host "Original dist-tag: $parsedOriginalDistTags"
+Write-Host "Current dist-tag: $packageDistTags"
+Write-Host "Intend to add tag $intendedTag to version $intendedTagVersion"
+
+if ($packageDistTags."$intendedTag" -ne $intendedTagVersion) {
+  Write-Warning "Tag not correctly set, current $intendedTag tag is version $($packageDistTags."$intendedTag") instead of $intendedTagVersion."
+  $correctDistTags = $parsedOriginalDistTags
+  $correctDistTags."$intendedTag" = $intendedTagVersion
+
+  Write-Host "Setting AuthToken Deployment"
+  $regAuth = "//registry.npmjs.org/"
+  $env:NPM_TOKEN=$npmToken
```

Comment:
Does this need to be in the environment? If so we can just set it as and env variable in the devops step.